### PR TITLE
feat: adoc patch --apply — formatting-preserving patch application (V6.4)

### PR DESCRIPTION
**Pull request by alex-bako**
## Summary

Closes the write half of the agent editing loop (ROADMAP-V6 §V6.4, ADR-0036, ADR-0037): every already-validated patch op family (`update_fields`, `replace_body`, `supersede`, `revoke`, `create_object`) now actually rewrites `.adoc` source via formatting-preserving span splices — atomic temp+fsync+rename writes, an automatic post-apply re-check embedded in the envelope, and never an auto-revert. Git stays the review and rollback mechanism.

### Tracer bullets

- **TB1 — splice engine + `update_fields`/`replace_body`**: close-fence span retention on `ParsedTypedBlock`, pure `domain/source_edit/` (`SpanEdit`, `SourceEditPlan`, `splice()`), `application/apply.rs` orchestration, `WorkspaceWriter` port, CLI `adoc patch --apply <path-or-@-stdin>` with exit codes `0` applied+clean / `1` refused / `2` applied-but-post-check-errors.
- **TB2 — relation ops**: `supersede` and `revoke` apply as field-line edits under the same splice discipline.
- **TB3 — `create_object` placement**: `PlacementHint` apply semantics (`page_id` → file, `after` → insert past close fence, absent → append); new diagnostics `patch.create_missing_placement` (WARNING on check, ERROR on apply) and `patch.placement_not_adoc`.
- **TB4 — gated MCP `adoc_patch_apply`**: registered always, refuses unless `mcp: { patch_apply: enabled }` in `agentdoc.config.yaml`; `patch_apply_enabled` readiness on `adoc.project.status.v0`; `adoc://agent/v0/patch-apply-guide` + JSON Schema resources; apply-aware `adoc_propose_patch_v1` prompt (v0 byte-stable per ADR-0014); every "MCP does not apply patches" doc promise rescoped to validation/review.
- **TB5 — Expanded Pilot full-loop proof** (`crates/adoc-cli/tests/apply_loop.rs`): `impacted-by` → propose → apply (golden-file byte-exact) → post-check clean → `stale`/`contradictions` unchanged → re-apply refuses on stale `base_hash` with no write.

### Key invariants

- **Two-layer freshness**: existing `base_hash` check (proposer saw the artifact) plus the new apply-time `patch.source_drift` gate (in-memory recompile must reproduce the artifact's `content_hash`).
- **Splicing never trusts artifact spans**: apply re-parses current source and uses byte offsets (`SourcePosition.offset`) only; property tests pin empty-plan byte-identity, outside-range preservation, and multibyte char-boundary safety.
- **Artifacts go stale loudly**: apply never rewrites `dist/`; the `adoc.patch.apply.v0` envelope carries `artifacts_stale: true`.

New wire envelope `adoc.patch.apply.v0` is JSON-Schema'd and contract-tested (applied, gate-refusal, and base-hash-refusal cases); `adoc.patch.v0` stays at v0.

### Also included

- V6.3 `impacted-by` follow-up polish (markdown error rendering, empty-paths MCP argument guard, obligation emission refactor).
- fix(core): centralize the fastembed model cache to a stable directory.

## Test plan

- [x] `cargo test` workspace-wide (unit + property tests in `domain/source_edit/`, planner edge cases)
- [x] `crates/adoc-cli/tests/apply_loop.rs` — full-loop Expanded Pilot proof with golden-file byte comparison
- [x] `crates/adoc-cli/tests/patch_cli.rs` — `--apply` CLI surface, placement refusals
- [x] `crates/adoc-mcp/tests/contract_schemas.rs` / `mcp_adapter.rs` — envelope schema validation, gate posture

### CODE REVIEW

File: crates/adoc-core/src/infrastructure/source/fs_writer.rs [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        if sha256_prefixed(&current) != expected_current_hash {
+            let _ = std::fs::remove_file(&temp_path);
+            return Err(WorkspaceWriteError::ConcurrentModification { path: resolved });
+        }
+
+        std::fs::rename(&temp_path, &resolved).map_err(|error| {
+            let _ = std::fs::remove_file(&temp_path);
+            io_error(&resolved, error)
+        })
+    }
```
[the remark this answers is not in the document]
Comment (alex-bako) [reply crates/adoc-core/src/infrastructure/source/fs_writer.rs:150]: Fixed in d3416d3: best-effort `File::open(directory).sync_all()` after the successful rename. Reported as best-effort deliberately — the write itself already succeeded, so surfacing a directory-sync failure as an apply failure would misreport the file's contents. Module docstring updated to name the parent-directory sync.

[the remark this answers is not in the document]
Comment (alex-bako) [reply crates/adoc-core/src/infrastructure/source/fs_writer.rs:150]: Two parts to this one. The steady-state case was already correct: `contained()` canonicalizes an existing target per call, so a symlink present when `write_atomic` starts resolves to the real file — the read, hash check, and rename all act on the link's target and the link survives. Now pinned by `write_atomic_writes_through_a_preexisting_symlink`.

What remained was the swap race you describe. a7118b5 adds a `symlink_metadata` check immediately before the rename that refuses with `ConcurrentModification` when a symlink is observed (portable — no Linux-only `O_NOFOLLOW`/`renameat2`). The residual instant between that check and `rename(2)` is now explicitly scoped in the `WorkspaceWriter` port docs next to the cross-process-locking non-goal, since no deterministic test can open that window without test-only seams.

File: crates/adoc-core/src/domain/source_edit/planner.rs [hunk 2]
```diff
@@ -0,0 +1,668 @@
...
+        output.push_str(": ");
+        output.push_str(value);
+        output.push_str(eol);
+    }
+    output.push_str("--");
+    output.push_str(eol);
+    for line in body.split('\n') {
+        output.push_str(line.strip_suffix('\r').unwrap_or(line));
+        output.push_str(eol);
+    }
```
[the remark this answers is not in the document]
Comment (alex-bako) [reply crates/adoc-core/src/domain/source_edit/planner.rs]: Fixed in c08fa0e, and the defect was broader than reported: `join_lines` (the `replace_body` path) had the same `split('\n')` trailing-empty-piece behavior. Both now route through `trim_one_trailing_newline`, which strips exactly one trailing line terminator (`\n` or `\r\n`) before splitting — so `"a\n"` and `"a"` render byte-identically, while a deliberate trailing blank line (`"a\n\n"`) keeps exactly one blank line. Idempotency pinned by tests on both `render_typed_block` and `plan_replace_body`.

File: crates/adoc-local/src/config.rs [hunk 3]
```diff
@@ -13,6 +13,13 @@ pub struct ProjectConfig {
     pub docs_path: PathBuf,
     pub outputs: ConfigOutputs,
     pub embeddings_provider: EmbeddingsProvider,
+    /// V6.4 TB4 (ADR-0037): the MCP `adoc_patch_apply` gate. Absent `mcp:`
+    /// block ⇒ disabled; `adoc init` never writes the key — opting in is a
+    /// deliberate human edit. Note the back-compat consequence of
+    /// `deny_unknown_fields`: a project that adds the `mcp:` block becomes
+    /// unreadable by pre-V6.4 binaries (a loud config-parse failure that
+    /// only bites opted-in projects; deliberately no version bump).
+    pub mcp_patch_apply_enabled: bool,
```
[the remark this answers is not in the document]
Comment (alex-bako) [reply crates/adoc-local/src/config.rs:22]: Keeping as-is — this was decided deliberately in ADR-0037 §8 ("Config back-compat, documented not versioned"): the failure is loud, only bites projects that opted in, and the config `version` is deliberately not bumped.

On the two softenings: loosening top-level `deny_unknown_fields` means a typo like `mpc:` silently disables a write gate instead of failing loudly — for a key that gates filesystem writes, a loud parse error is the safer failure mode. A version bump forces the ceremony on every project including the majority that never opts in. The blast radius is limited to opted-in projects running mixed toolchains, and "upgrade adoc" is the fix either way; the parse error names the unknown `mcp` key, which points there.
